plugins/org.eclipse.n4js/src/org/eclipse/n4js/xtext/workspace/ProjectBuildOrderInfo.java

```diff
@@ -125,36 +121,18 @@ public ProjectConfigSnapshot next() {
 		}
 	}
 
-	/** Workspace manager */
-	@Inject
-	protected XWorkspaceManager workspaceManager;
-
+	/**  */
+	protected WorkspaceConfigSnapshot workspaceConfig;
 	/** Inverse set of project dependency information */
 	final protected Multimap<String, ProjectConfigSnapshot> inversedDependencies = HashMultimap.create();
 	/** Build order of projects */
 	final protected List<ProjectConfigSnapshot> sortedProjects = new ArrayList<>();
-
-	/**
-	 * Creates a new instance of {@link ProjectBuildOrderIterator}. Assumes a succeeding call to
-	 * {@link ProjectBuildOrderIterator#visit(Collection)} method.
-	 */
-	@Override
-	public ProjectBuildOrderIterator getIterator() {
-		return new ProjectBuildOrderIterator();
-	}
-
-	/** Creates a new instance of {@link ProjectBuildOrderIterator}. The given set of projects will be visited only. */
-	@Override
-	public ProjectBuildOrderIterator getIterator(Collection<? extends ProjectConfigSnapshot> projectDescriptions) {
-		ProjectBuildOrderIterator iterator = getIterator();
-		iterator.visit(projectDescriptions);
-		return iterator;
-	}
+	/** All project cycles, each cycle given as a list of project names */
+	final protected Collection<List<String>> projectCycles = new HashSet<>();
 
 	/** Populates {@link #sortedProjects} and {@link #inversedDependencies} */
```

Comment:
Does this method now also populate field `#projectCycles`? If so, please add this to API doc.